src/app/services/auth.service.ts

```diff
@@ -0,0 +1,72 @@
+import {Injectable} from '@angular/core';
+import {Router} from '@angular/router';
+import {auth} from 'firebase/app';
+import {Observable} from 'rxjs';
+import {User} from '../models/user.model';
+import * as firebase from 'firebase';
+import {NotificationService} from './notification.service';
+import {of} from 'rxjs/internal/observable/of';
+import {environment} from '../../environments/environment';
+import {AngularFireAuth} from '@angular/fire/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  user$: Observable<User | null | undefined> = of(undefined);
+
+  constructor(
+    private router: Router,
+    private notificationService: NotificationService,
+    public angularFireAuth: AngularFireAuth
+  ) {
+    if (environment.requireAuth) {
+        angularFireAuth.onAuthStateChanged(firebaseUser => {
+          if (firebaseUser != null) {
+            const user: User = {uid: firebaseUser.uid, displayName: firebaseUser.displayName};
+            this.user$ = of(user);
+          } else {
+            this.user$ = of(null);
+          }
+        });
+      }
+  }
+
+  googleSignin(): void {
+    const provider = new auth.GoogleAuthProvider();
+    this.angularFireAuth.signInWithPopup(provider).then(result => {
+      const user = result.user;
+      if (user == null) {
+        throw new Error('User is null');
+      } else {
+        this.notificationService.notification$.next('Signed in as ' + user.displayName);
+        this.router.navigate(['/']);
+        return this.updateUserData(user);
+      }
+    }).catch(error => {
+      console.log(error);
```

Comment:
Maybe we should send this to your notification service as an error, so we can make a central decision about how we want to display errors?

Reply:
At the very least, I think these should use `console.error` so they show up red in the console.

Reply:
I changed it to
`this.notificationService.error$.next(error);`
in my last push, not sure if you saw that, I assume that is appropriate?

Reply:
My mistake, sorry - I assumed because this code snippet was still here in the review, that this code was still on the head of your branch.